### PR TITLE
feat: add a sample to show how to resolve expression for multiple records

### PR DESCRIPTION
**Pull request by qiqingjin**
image

### CODE REVIEW

File: widgets/use-expression/src/runtime/widget.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  const renderExpressionResult = (resolvedResults: SingleExpressionResolveResult) => {
+    if (!resolvedResults?.isSuccessful) {
+      return null
+    }
+    return <div className='record-item'>{resolvedResults.value}</div>
+  }
+
+  const renderData = () => {
+    const dsIdUsedInExpression = expression.parts?.find(p => p.dataSourceId)?.dataSourceId
```
Comment (qlqllu) [inline widgets/use-expression/src/runtime/widget.tsx:19]: The renderData method should have been passed in the data source instance.

**Comment by qlqllu** [comment]
Hi @gavinr we plan to add this sample widget in the coming release to demo how to use the expression, could you give it a review?

File: widgets/use-expression/src/runtime/widget.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    </div>
+  }
+
+  return (
+    <div className='widget-use-expression jimu-widget' css={style}>
+      {
+        useDataSources?.[0] && expression
+          ? <DataSourceComponent useDataSource={useDataSources[0]} query={DEFAULT_QUERY} widgetId={id}>{renderData}</DataSourceComponent>
+          : <div>
+              <p>This widget demostrates how to resolve expression for multiple records.</p>
```
Comment (qiqingjin) [inline widgets/use-expression/src/runtime/widget.tsx:52]:
```suggestion
              <p>This widget demonstrates how to resolve expression for multiple records.</p>
```

**Comment by qiqingjin** [comment]
Hi @gavinr, the sample widget is ready to review, could you help to take a look?

This widget demonstrates how to resolve expression for multiple records. It is from a user question. https://community.esri.com/t5/arcgis-experience-builder-questions/how-to-get-expressionresolvercomponent-to-resove/m-p/1197248 .

**Comment by gavinr** [comment]
@qiqingjin can you please post the recommended way to configure an application to use this demo widget? I tried my best but it shows a selected item 100 times in the list which I'm not sure if that correct or not:

image

File: widgets/use-expression/src/runtime/widget.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  )
+}
+
+export default Widget
+
+const style = css`
+  height: 100%;
+  overflow: auto;
+  .record-item {
+    border: 1px solid blue;
```
Comment (qiqingjin) [inline widgets/use-expression/src/runtime/widget.tsx:66]:
```suggestion
    border: 1px solid gray;
```

File: widgets/use-expression/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+# Use expression
+
+This sample demonstrates how to resolve expression for multiple records in a custom widget.
```
Comment (qiqingjin) [inline widgets/use-expression/README.md:3]:
```suggestion
This sample demonstrates how to resolve expression for multiple records in a custom widget.

In setting panel, select a data source and add an expression. In widget, you will see the expression is resolved to value. Please note the sample will only load the first page (100 records by default).
```

**Pull request merged**